charts/templates/secret.yaml

```diff
@@ -1,11 +1,20 @@
+{{- $adminPassword := (randAlphaNum 10) | b64enc | quote }}
+{{- $secret := (lookup "v1" "Secret" .Release.Namespace .Release.Name ) }}
+{{- if $secret }}
+{{- $adminPassword = index $secret.data "password" }}
+{{- end -}}
+
 apiVersion: v1
 kind: Secret
 metadata:
   name: {{ include "marklogic.fullname" . }}
   namespace: {{ .Release.Namespace }}
   labels:
     {{- include "marklogic.labels" . | nindent 4 }}
-type: kubernetes.io/basic-auth
-stringData:
-  username: {{ .Values.auth.adminUsername}}
-  password: {{ .Values.auth.adminPassword}}
\ No newline at end of file
+type: Opaque
```

Comment:
Couldn't we keep the type: kubernetes.io/basic-auth as it is more convenient than Opaque

Reply:
kubernetes.io/basic-auth is very similar to Opaque except that it requires to have the key of username and password. I have 2 reasonings of change from basic-auth to Opaque:
1. this secret will be used not only for username and password. It could also be used for storing ssh keys and other possible password in the future.
2. the username and password is too generic, I'd like to change it to marklogic-username and marklogic-password, which violate the requirement for basic-auth

Reply:
Idea here is to have things clear from an admin perspective.

1. Having dedicated secret for each stuff is a good practice.
2. Changing the secret name would make thing more clear e.g name: {{ include "marklogic.fullname" . }}-admin

Reply:
Sure. I've made a PR to apply these changes.